### PR TITLE
lazy_load_no_trans init commit

### PR DESCRIPTION
**Pull request by petershatunov**

### CODE REVIEW

File: persistence-modules/spring-boot-persistence-h2/src/main/java/com/baeldung/h2db/lazy_load_no_trans/service/ServiceLayer.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public int countAllDocsNonTransactional() {
+        return countAllDocs();
+    }
+
+    private int countAllDocs() {
+        List<Document> docs = userRepository
+            .findAll().stream()
+            .flatMap(user -> user.getDocs().stream())
+            .collect(Collectors.toList());
+        return docs.size();
```
Comment (ashleyfrieze) [inline persistence-modules/spring-boot-persistence-h2/src/main/java/com/baeldung/h2db/lazy_load_no_trans/service/ServiceLayer.java]: You don't need to collect to a list and then return the size of the list. You can go

```java
return userRepository.findAll()
    .stream()
    .map(User::getDocs)
    .flatMap(Document::stream)
    .count();
```

Or similar.

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Test(expected = LazyInitializationException.class)
+    public void should_call_non_transactional_method() {
+        serviceLayer.countAllDocsNonTransactional();
+    }
+
+    @Test
+    public void should_call_transactional_method() {
+        SQLStatementCountValidator.reset();
+        long docsCount = serviceLayer.countAllDocsTransactional();
+        Assert.assertEquals(EXPECTED_DOCS_COLLECTION_SIZE, docsCount);
```
Comment (ashleyfrieze) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java]: Static import `assertEquals`

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            .countQuery()
+            .listener(loggingListener)
+            .build();
+    }
+
+    @Test
+    public void should_call_non_transactional_method() {
+        SQLStatementCountValidator.reset();
+        long docsCount = serviceLayer.countAllDocsNonTransactional();
+        Assert.assertEquals(EXPECTED_DOCS_COLLECTION_SIZE, docsCount);
```
Comment (ashleyfrieze) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]: static import `assertEquals`

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        SQLStatementCountValidator.assertSelectCount(EXPECTED_USERS_COUNT + 1);
+    }
+
+    @Test
+    public void should_call_transactional_method() {
+        SQLStatementCountValidator.reset();
+        long docsCount = serviceLayer.countAllDocsTransactional();
+        Assert.assertEquals(EXPECTED_DOCS_COLLECTION_SIZE, docsCount);
```
Comment (ashleyfrieze) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]: and here

Comment (petershatunov) [reply persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]: fixed

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java (hunk 3 above)
Comment (petershatunov) [reply persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]: fixed

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java (hunk 2 above)
Comment (petershatunov) [reply persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java]: fixed

File: persistence-modules/spring-boot-persistence-h2/src/main/java/com/baeldung/h2db/lazy_load_no_trans/service/ServiceLayer.java (hunk 1 above)
Comment (petershatunov) [reply persistence-modules/spring-boot-persistence-h2/src/main/java/com/baeldung/h2db/lazy_load_no_trans/service/ServiceLayer.java]: That's right, fixed

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    private static final long EXPECTED_DOCS_COLLECTION_SIZE = 6;
+
+    @Test(expected = LazyInitializationException.class)
+    public void whenCallNonTransactionalMethodWithPropertyOffThenThrowException() {
+        serviceLayer.countAllDocsNonTransactional();
+    }
+
+    @Test
+    public void whenCallTransactionalMethodWithPropertyOffThenTestPass() {
```
Comment (KevinGilmore) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOffIntegrationTest.java]:
```suggestion
    public void whenCallTransactionalMethodWithPropertyOff_thenTestPass() {
```

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+public class LazyLoadNoTransPropertyOnIntegrationTest {
+
+    @Autowired
+    private ServiceLayer serviceLayer;
+
+    private static final long EXPECTED_DOCS_COLLECTION_SIZE = 6;
+    private static final long EXPECTED_USERS_COUNT = 5;
+
+    @Test
+    public void whenCallNonTransactionalMethodWithPropertyOnThenGetNplusOne() {
```
Comment (KevinGilmore) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]:
```suggestion
    public void whenCallNonTransactionalMethodWithPropertyOn_thenGetNplusOne() {
```

File: persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        SQLStatementCountValidator.reset();
+        long docsCount = serviceLayer.countAllDocsNonTransactional();
+        assertEquals(EXPECTED_DOCS_COLLECTION_SIZE, docsCount);
+        SQLStatementCountValidator.assertSelectCount(EXPECTED_USERS_COUNT + 1);
+    }
+
+    @Test
+    public void whenCallTransactionalMethodWithPropertyOnThenTestPass() {
```
Comment (KevinGilmore) [inline persistence-modules/spring-boot-persistence-h2/src/test/java/com/baeldung/lazy_load_no_trans/LazyLoadNoTransPropertyOnIntegrationTest.java]:
```suggestion
    public void whenCallTransactionalMethodWithPropertyOn_thenTestPass() {
```

**Pull request merged**